recipes-asteroid/qml-asteroid/qml-asteroid_git.bb

```diff
@@ -11,34 +11,17 @@ PV = "+git${SRCPV}"
 S = "${WORKDIR}/git"
 inherit qmake5
 
-PACKAGE_ARCH = "${MACHINE_ARCH}"
-
 DEPENDS += "qtdeclarative qtsvg qtvirtualkeyboard mlite mapplauncherd-booster-qtcomponents qtdeclarative-native"
 RDEPENDS_${PN} += "qtsvg-plugins qtvirtualkeyboard asteroid-icons-ion"
 
-do_configure_prepend() {
-    if [ ${MACHINE_DISPLAY_FLAT_TIRE} ]
-    then
-        export EXTRA_QMAKEVARS_PRE="${EXTRA_QMAKEVARS_PRE} DEFINES+=FLAT_TIRE=${MACHINE_DISPLAY_FLAT_TIRE}"
-    fi
-    if [ ${MACHINE_DISPLAY_BORDER_GESTURE_WIDTH} ]
-    then
-        export EXTRA_QMAKEVARS_PRE="${EXTRA_QMAKEVARS_PRE} DEFINES+=BORDER_GESTURE_WIDTH=${MACHINE_DISPLAY_BORDER_GESTURE_WIDTH}"
-    fi
-    if [ ${MACHINE_DISPLAY_ROUND} = "true" ]
-    then
-        export EXTRA_QMAKEVARS_PRE="${EXTRA_QMAKEVARS_PRE} DEFINES+=ROUND_SCREEN"
-    fi
-    if [ ${MACHINE_HAS_WLAN} = "true" ]
-    then
-        export EXTRA_QMAKEVARS_PRE="${EXTRA_QMAKEVARS_PRE} DEFINES+=HAS_WLAN"
-    fi
-    if [ ${MACHINE_HAS_SPEAKER} = "true" ]
-    then
-        export EXTRA_QMAKEVARS_PRE="${EXTRA_QMAKEVARS_PRE} DEFINES+=HAS_SPEAKER"
+do_install_append() {
+    if [ -f ../machine.conf ]; then
+      install -d ${D}/etc/
+      cp ../machine.conf ${D}/etc/machine.conf
     fi
 }
 
-FILES_${PN} += "/usr/lib /usr/share/icons/asteroid/"
+
+FILES_${PN} += "/etc/ /usr/lib /usr/share/icons/asteroid/"
```

Comment:
I think the configuration file should be distributed as a separate opkg. It makes sense for qml-asteroid to be one package that can be installed on any watch but not the config file.